src/scrapers/mizrahi.js

```diff
@@ -0,0 +1,119 @@
+import moment from 'moment';
+import {
+  SHEKEL_CURRENCY,
+  NORMAL_TXN_TYPE,
+  TRANSACTION_STATUS,
+  ISO_DATE_FORMAT,
+} from '../constants';
+import { BaseScraperWithBrowser, LOGIN_RESULT } from './base-scraper-with-browser';
+import { fetchPostWithinPage } from '../helpers/fetch';
+
+const BASE_URL = 'https://www.mizrahi-tefahot.co.il';
+const LOGIN_URL = `${BASE_URL}/he/bank/Pages/Default.aspx`;
+const AFTER_LOGIN_BASE_URL = 'https://mto.mizrahi-tefahot.co.il/ngOnline/index.html#/main/uis/osh/p428/';
+const TRANSACTIONS_REQUEST_URL = 'https://mto.mizrahi-tefahot.co.il/Online/api/SkyOSH/get428Index';
+const DATE_FORMAT = 'DD/MM/YYYY';
+
+function createLoginFields(credentials) {
+  return [
+    { selector: '#ctl00_PlaceHolderLogin_ctl00_tbUserName', value: credentials.username },
+    { selector: '#ctl00_PlaceHolderLogin_ctl00_tbPassword', value: credentials.password },
+  ];
+}
+
+function getPossibleLoginResults() {
+  const urls = {};
+  urls[LOGIN_RESULT.SUCCESS] = [`${AFTER_LOGIN_BASE_URL}`];
+  urls[LOGIN_RESULT.INVALID_PASSWORD] = [`${BASE_URL}/login/loginMTO.aspx`];
+  urls[LOGIN_RESULT.CHANGE_PASSWORD] = [
+    `${AFTER_LOGIN_BASE_URL}/main/uis/ge/changePassword/`,
+  ];
+  return urls;
+}
+
+function CreateDataFromRequest(request, optionsStartDate) {
+  const defaultStartMoment = moment().subtract(1, 'years');
+  const startDate = optionsStartDate || defaultStartMoment.toDate();
+  const startMoment = moment.max(defaultStartMoment, moment(startDate));
+
+  const data = JSON.parse(request.postData());
+
+  data.inToDate = moment().format(DATE_FORMAT);
+  data.inFromDate = startMoment.format(DATE_FORMAT);
+  data.table.maxRow = 9999999999;
```

Comment:
is this a valid value that they send as well? if not it might annoy their developers, consider changing to the a bit more reasonable number. If they also send this value feel free to ignore this comment.

Reply:
They send 50, because this is their page size for paging.
I tried and reached that number. approximately. When it is bigger the request fails.

I hope there is no user with more transactions per year, otherwise we will have to make another request.